### DISCUSSION TITLE
How to run again?

### DISCUSSION THREAD

**Discussion by rastarr** [mrshmllow/affinity-nix#9] [Q&A] [answered]
thanks so much for this repo.
i managed to install affinity photo. 
once you quit the app, how do you run it again though?
I'm not seeing it in my i3 run menu so a bit confused and a total newbie at Wine

**Comment by mrshmllow** [comment]
Hey hey, sorry ive been busy and unable to respond to github. How did you install the application? If you _install_ it through nix the .desktop files should exist. If not, it may be a bug in this repo.

The code examples just run it one-shot

**Comment by rastarr** [comment]
it's ok now and thanks for the reply.
The affinity apps are too flakey and unstable for my liking. I'll continue using a Windows VM for Affinity and Itunes.
Maybe one bright sunny day, Affinity might come to native Linux but I won't hold my breathe :)

**Comment by mrshmllow** [comment]
Yep I feel the same tbh

**Comment by emenel** [comment]
i realize that it might be flaky, but i'd still love to understand how to install these in my nixos/home-manager system if possible. i've tried a few different things but so far nothing has worked.

thanks for all the work on this!

**Comment by mrshmllow** [comment, marked as answer]
> i realize that it might be flaky, but i'd still love to understand how to install these in my nixos/home-manager system if possible. i've tried a few different things but so far nothing has worked.

You can use nix profile install `nix profile install github:mrshmllow/affinity-nix#photo`

Otherwise, add this repo as a flake input, then just access `affinity-nix.packages.x86_64-linux.photo` anywhere you would install a package.

```nix
{
  inputs = {
    affinity-nix.url = "github:mrshmllow/affinity-nix";
    ...
  };

  outputs = inputs @ {
    affinity-nix,
    ...
  }: {
    homeConfigurations.marsh = home-manager.lib.homeManagerConfiguration {
      pkgs = nixpkgs.legacyPackages."x86_64-linux";
      extraSpecialArgs = {inherit inputs;};
      modules = [
        {
          home.packages = [affinity-nix.packages.x86_64-linux.photo];
        }
      ];
    };
  }
}
```

this applies to every flake that exposes a package :)

if you want to run the other scripts like `updatePhoto`, use `nix run` like in the readme

**Reply by emenel** [reply]
Thanks for the detailed answer!

**Reply by emenel** [reply]
One more followup, I'm not getting an error about an unfree license when it gets to setting up Affinity Photo. I have allowUnfree set in my config and am currently using other unfree packages. I'm not sure if I'm missing something basic again, or something else.

Thanks again, even for the general pointers. A lot of these features and patterns are very under-documented or out of date, this really helps me understand importing flake packages in general much more clearly.

**Reply by corbinwunderlich** [reply]
> One more followup, I'm not getting an error about an unfree license when it gets to setting up Affinity Photo. I have allowUnfree set in my config and am currently using other unfree packages. I'm not sure if I'm missing something basic again, or something else.
> 
> Thanks again, even for the general pointers. A lot of these features and patterns are very under-documented or out of date, this really helps me understand importing flake packages in general much more clearly.

Please fix this, I am having the same issue :(

**Reply by corbinwunderlich** [reply]
@mrshmllow

**Reply by mrshmllow** [reply]
> Please fix this, I am having the same issue :(

I am not entirely sure what the issue is. I think you should start a new thread with the details nevertheless @CorbinWunderlich
